app/src/main/java/com/android/streetworkapp/model/park/ParkViewModel.kt

```diff
@@ -57,7 +57,7 @@ open class ParkViewModel(private val repository: ParkRepository) : ViewModel() {
   fun getParkByPid(pid: String) {
     viewModelScope.launch {
       val fetchedPark = repository.getParkByPid(pid)
-      _park.value = fetchedPark
+      _currentPark.value = fetchedPark
```

Comment:
Question: I was not there when @SaturneV designed the MVVM and I guess he must have explained the difference between a `park` and a `currentPark`. 
With this change I'm a bit lost, I see only two possibilities: 

1. Either the `currentPark` and the `park` make sense as in MVVM (which means that there are cases where you need to have both at the same time), or the `currentPark` and the `park` make sense as in MVVM (which means that there are cases where you need to have both at the same time). In this case, certain functions need to be added to make `currentPark` behave like `park`.
2. Or delete `park` and work only with `currentPark` in this MVVM (because you never need both at the same time).

Could you enlighten me on this question?

Reply:
The reason for this change is mainly because of functionality issues with the park overview:
Essentially i need to update two different values to make it work properly.
If you look at the map ui, the `getOrCreatePark()` function is called when clicked on a marker to set the `Park` value and then the `getParkByPid()` function is called in the park overview. 
It is necessary to set the current park in the park overview for two reasons:
- first is because the park overview could be accessed in other ways than the map, for  example you return to the park overview after creating an event. If `getParkByPid()` wasn't called at this point, events would not be updated a new one has been added to firebase
- The second reason is for consistency issues with the park overview: if the current park is set to different values in values in quick succesion (i.e in the map ui and park overview) there seems to be weird asynchronous behavior where the values might not be set in the right order and events might not be displayed properly. This is the reason i am using both values for my park overview and why i changed the set value of `getParkByPid`

I acknowledge that it might not entirely be clear when looking at the code but this is the best way i have found to solve these issues. Let me know if you think there is a better way to resolve this problem

Reply:
I'm not sure I fully understood your answer so I'll try to rephrase it. And then I'll show you what bothers me about your answer.

You need currentpark and park, for example in mapUI where you have several parks to display.
But what I still don't understand is why modify the `getParkByPid` function the way you did.
I don't think you've seen it, but in the MVVM that @SaturneV has implemented, the `loadCurrentPark` function does exactly what you want it to do! 

To be more formal, I'll go into my problem description.
In your PR you change the `getParkByPid` function:
1. Before your PR
```kotlin
fun getParkByPid(pid: String) {
    viewModelScope.launch {
      val fetchedPark = repository.getParkByPid(pid)
      _park.value = fetchedPark
      }
  }
```

2. After your PR
```kotlin
fun getParkByPid(pid: String) {
    viewModelScope.launch {
      val fetchedPark = repository.getParkByPid(pid)
      _currentPark.value = fetchedPark
    }
  }
```

3. But `loadCurrentPark` is the exact same function as your modified `getParkByPid` function.
```kotlin
fun loadCurrentPark(pid: String) {
    viewModelScope.launch {
      val fetchedPark = repository.getParkByPid(pid)
      _currentPark.value = fetchedPark
    }
  }
```
There's no world in which 2 and 3 can coexist in the same MVVM, do you agree ?

My suggestion is that you use `loadCurrentPark` where you used to use `getParkByPid` and go back to the version before the changes to the `getParkByPid` function.

Reply:
you are right, i did not realise that there already was a function doing the exact same thing, I have applied the suggestion
Thank you for pointing that out